stdlib/public/core/Span/Span.swift

```diff
@@ -0,0 +1,827 @@
+  /// contain equal elements in the same order.
+  ///
+  /// - Parameters:
+  ///   - other: A Sequence to compare to this span.
+  /// - Returns: `true` if this sequence and `other` contain equivalent items,
+  ///   using `areEquivalent` as the equivalence test; otherwise, `false.`
+  ///
+  /// - Complexity: O(*m*), where *m* is the lesser of the length of the
+  ///   sequence and the length of `other`.
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  public func _elementsEqual(_ other: some Sequence<Element>) -> Bool {
+    var offset = 0
+    for otherElement in other {
+      if offset >= count { return false }
+      if self[unchecked: offset] != otherElement { return false }
+      offset += 1
+    }
+    return offset == count
+  }
+}
+
+@_disallowFeatureSuppression(NonescapableTypes)
+@available(SwiftStdlib 9999, *)
+extension Span where Element: ~Copyable {
+
+  /// The number of elements in the span.
+  ///
+  /// To check whether the span is empty, use its `isEmpty` property
+  /// instead of comparing `count` to zero.
+  ///
+  /// - Complexity: O(1)
+  @_alwaysEmitIntoClient
+  public var count: Int { _count }
+
+  /// A Boolean value indicating whether the span is empty.
+  ///
+  /// - Complexity: O(1)
+  @_alwaysEmitIntoClient
+  public var isEmpty: Bool { _count == 0 }
+
+  /// The representation for a position in `Span`.
+  public typealias Index = Int
+
+  /// The indices that are valid for subscripting the span, in ascending
+  /// order.
+  ///
+  /// - Complexity: O(1)
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  public var indices: Range<Index> {
+    Range(_uncheckedBounds: (0, _count))
+  }
+}
+
+//MARK: integer offset subscripts
+@_disallowFeatureSuppression(NonescapableTypes)
+@available(SwiftStdlib 9999, *)
+extension Span where Element: ~Copyable {
+
+  /// Accesses the element at the specified position in the `Span`.
+  ///
+  /// - Parameter position: The offset of the element to access. `position`
+  ///     must be greater or equal to zero, and less than `count`.
+  ///
+  /// - Complexity: O(1)
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @_alwaysEmitIntoClient
+  public subscript(_ position: Index) -> Element {
+    _read {
+      _precondition(indices.contains(position), "index out of bounds")
+      yield self[unchecked: position]
+    }
+  }
+
+  /// Accesses the element at the specified position in the `Span`.
+  ///
+  /// This subscript does not validate `position`; this is an unsafe operation.
+  ///
+  /// - Parameter position: The offset of the element to access. `position`
+  ///     must be greater or equal to zero, and less than `count`.
+  ///
+  /// - Complexity: O(1)
+  @_disallowFeatureSuppression(NonescapableTypes)
+  @unsafe
+  @_alwaysEmitIntoClient
+  public subscript(unchecked position: Index) -> Element {
+    //FIXME: change to unsafeRawAddress or unsafeAddress when ready
```

Comment:
I assume the same FIXME applies to the index subscript above.

Reply:
It does! I'll add it there too.